public/app/features/alerting/unified/utils/datasource.ts

```diff
@@ -363,3 +348,45 @@ export function ruleIdentifierToRuleSourceIdentifier(ruleIdentifier: RuleIdentif
     ruleSourceType: 'datasource',
   };
 }
+
+/**
+ * Check if the given type is a supported external Prometheus flavored rules source type.
+ */
+export function isSupportedExternalPrometheusFlavoredRulesSourceType(
+  type: string
+): type is SupportedExternalPrometheusFlavoredRulesSourceType {
+  return SUPPORTED_EXTERNAL_PROMETHEUS_FLAVORED_RULE_SOURCE_TYPES.includes(
+    type as SupportedExternalPrometheusFlavoredRulesSourceType
+  );
+}
+export const SUPPORTED_EXTERNAL_PROMETHEUS_FLAVORED_RULE_SOURCE_TYPES = [
+  'prometheus',
+  'grafana-amazonprometheus-datasource',
+  'grafana-azureprometheus-datasource',
+] as const;
+export type SupportedExternalPrometheusFlavoredRulesSourceType =
+  (typeof SUPPORTED_EXTERNAL_PROMETHEUS_FLAVORED_RULE_SOURCE_TYPES)[number]; // infer the type from the tuple above so we can maintain a single source of truth
```

Comment:
this might seem a bit odd but allows the type to infer the string union from the tuple defined above, that way we don't need to duplicate strings.